app/core/ingestion/family.py

```diff
@@ -0,0 +1,187 @@
+
+    :param [Session] db: connection to the database.
+    :param [int] org_id: the organisation id associated with this row.
+    :param [IngestRow] row: the row built from the CSV.
+    :param [dict[str, Any]] result: a result dict in which to track what was created
+    :raises [ValueError]: When there is an existing family name that only differs by
+        case or when the geography associated with this row cannot be found in the
+        database.
+    :return [Family]: The family that was either retrieved or created
+    """
+    # GET OR CREATE FAMILY
+    family = _maybe_create_family(db, row, org_id, result)
+
+    # GET OR CREATE FAMILY DOCUMENT
+    _maybe_create_family_document(db, row, family, existing_document, result)
+
+    return family
+
+
+def _maybe_create_family(
+    db: Session, row: IngestRow, org_id: int, result: dict[str, Any]
+) -> Family:
+    def _create_family_links(family: Family):
+        print(f"- Creating family slug for import {family.import_id}")
+        family_slug = Slug(name=row.cpr_family_slug, family_import_id=family.import_id)
+
+        db.add(family_slug)
+        result["family_slug"] = (to_dict(family_slug),)
+
+        print(f"- Creating family organisation for import {row.cpr_family_id}")
+        family_organisation = FamilyOrganisation(
+            family_import_id=family.import_id, organisation_id=org_id
+        )
+        db.add(family_organisation)
+        result["family_organisation"] = to_dict(family_organisation)
+
+        id, taxonomy = get_organisation_taxonomy(db, org_id)
+        add_metadata(db, cast(str, family.import_id), taxonomy, id, row)
+
+    category = FamilyCategory(row.category.upper())
+
+    # GET GEOGRAPHY
+    print(f"- Getting Geography for {row.geography_iso}")
+    geography = _get_geography(db, row)
+
+    if not _validate_family_name(db, row.family_name, row.cpr_family_id):
+        raise ValueError(
+            f"Processing row {row.row_number} got family {row.family_name} "
+            "that is different to the existing family title for family id "
+            f"{row.cpr_family_id}"
+        )
+    family = get_or_create(
+        db,
+        Family,
+        import_id=row.cpr_family_id,
+        extra={
+            "title": row.family_name,
+            "geography_id": geography.id,
+            "category_name": category,
+            "description": row.family_summary,
+            "family_status": FamilyStatus.PUBLISHED,
+        },
+        after_create=_create_family_links,
+    )
+    result["family"] = to_dict(family)
+    return family
+
+
+def _maybe_create_family_document(
+    db: Session,
+    row: IngestRow,
+    family: Family,
+    existing_document: Document,
+    result: dict[str, Any],
+) -> FamilyDocument:
+    print(f"- Creating family document for import {row.cpr_document_id}")
+
+    # FIXME: these should come from well-known values, not whatever is in the CSV
```

Comment:
Should we validate Variant by adding its allowed values to the metadata?  Or should this be an enum @joel-wright ?